src/ghastoolkit/octokit/codescanning.py

```diff
@@ -288,17 +302,26 @@ def getAnalyses(
     ) -> list[dict]:
         """Get a list of all the analyses for a given repository.
 
+        This function will retry X times with a Y second sleep between each retry to
+        make sure the analysis is ready.
+
         Permissions:
         - "Code scanning alerts" repository permissions (read)
 
         https://docs.github.com/en/enterprise-cloud@latest/rest/code-scanning#list-code-scanning-analyses-for-a-repository
         """
-        results = self.rest.get(
-            "/repos/{org}/{repo}/code-scanning/analyses",
-            {"tool_name": tool, "ref": reference or self.repository.reference},
-        )
-        if isinstance(results, list):
-            return results
+        counter = 0
```

Comment:
Maybe add one additional log on the final retry and none found to indicate the threshold was hit after x # of retries and log it as error? Something that would trigger the user to know they should go and fix this as it is not going to implement policy otherwise ... fail closed?